File: number1.ts
const m = 3;
const n = 3;
const s = "ULDDRRUU";

function startPosFinder(m: number, n: number, s: string): any {
    let x = 0;
    let y = 0;
    let rX = 0;
    let lX = 0;
    let uY = 0;
    let dY = 0;

    for (const i of s) {
        switch (i) {
            case 'D':
                y += 1;
                if (y > dY) dY = y;
                break;
            case 'U':
                y -= 1;
                if (y < uY) uY = y;
                break;
            case 'L':
                x -= 1;
                if (x < lX) lX = x;
                break;
            case 'R':
                x += 1;
                if (x > rX) rX = x;
                break;
        }
    }

    if (rX - lX >= m || dY - uY >= n) {
        return "No";
    } else {
        return [1 - lX, 1 - uY];
    }
}

console.log(startPosFinder(m, n, s));

File: swift/number1.swift
let m1 = 3
let n1 = 3
let s1 = "ULDDRRUU"

func startPosFinder(m: Int, n: Int, s: String) -> (Int, Int)? {
    var x = 0
    var y = 0
    var rX = 0
    var lX = 0
    var uY = 0
    var dY = 0

    for char in s {
        switch char {
        case "D":
            y += 1
            dY = max(y, dY)
        case "U":
            y -= 1
            uY = min(y, uY)
        case "L":
            x -= 1
            lX = min(x, lX)
        case "R":
            x += 1
            rX = max(x, rX)
        default:
            break
        }
    }

    if rX - lX >= m || dY - uY >= n {
        return nil
    }
    return (1 - lX, 1 - uY)
}

if let startPos = startPosFinder(m: m1, n: n1, s: s1) {
    print(startPos)
} else {
    print("No")
}
